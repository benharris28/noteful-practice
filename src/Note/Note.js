import React from 'react';
import Context from '../Context';

class Note extends React.Component {
    static contextType = Context

    render() {
    const { notes } = this.context
    const { name, modified, content } = this.props


        return (
            <div>
            <div>{name}</div>
            <div>{modified}</div>
            
            </div>
        )
    }
}

export default Note;