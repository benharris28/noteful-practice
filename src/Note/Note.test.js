import React from 'react';
import ReactDOM from 'react-dom';
import Note from './Note';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('Note Component', () => {
    const props = {
        id: 'abcd',
        name: 'test note',
        modified: '2020-01-30'
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Note />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders note given props', () => {
        const wrapper = shallow(<Note {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})