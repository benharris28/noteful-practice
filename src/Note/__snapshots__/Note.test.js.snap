// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Component renders note given props 1`] = `
<div>
  <div>
    test note
  </div>
  <div>
    2020-01-30
  </div>
</div>
`;
